File: VB/App_Code/TimeZoneDataHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Web

<DataObject(True)> _
Public Class TimeZoneDataHelper
    Public ReadOnly Property TimeZones() As List(Of TimeZoneInfo)
        Get
            Return TimeZoneInfo.GetSystemTimeZones().ToList()
        End Get
    End Property

    <DataObjectMethod(DataObjectMethodType.Select, True)> _
    Public Function GetTimeZones() As List(Of TimeZoneInfo)
        Return TimeZones
    End Function
End Class

File: CS/App_Code/TimeZoneDataHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Web;

[DataObject(true)]
public class TimeZoneDataHelper {
    public List<TimeZoneInfo> TimeZones {
        get { return TimeZoneInfo.GetSystemTimeZones().ToList(); }
    }

    [DataObjectMethod(DataObjectMethodType.Select, true)]
    public List<TimeZoneInfo> GetTimeZones() {
        return TimeZones;
    }
}